src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js

```diff
@@ -37,16 +37,41 @@ const TextInputWithDropdown = React.forwardRef(
       label,
       dropdownInstruction = null,
       className = '',
+      hideBorder = false,
       ...otherInputProps
     },
-    ref
+    forwardedRef
   ) => {
     const [activeOption, setActiveOption] = useState(-1);
     const [forceClosed, setForceClosed] = useState(false);
+    const dropdownRef = useRef(null);
+    const containerRef = useRef(null);
+
+    useEffect(() => {
+      const handleClickOutside = event => {
+        if (dropdownRef.current
+          && !dropdownRef.current.contains(event.target)
+          && !containerRef.current.contains(event.target)) {
+          setForceClosed(true);
+        }
+      };
+
+      // Only add the listeners if we have options showing
+      if (options.length > 0 && !forceClosed) {
+        document.addEventListener('mousedown', handleClickOutside);
+        document.addEventListener('touchstart', handleClickOutside);
```

Comment:
If we wanted to get real fancy and streamlined

`["mousedown", "touchstart"].forEach(event => document.addEventListener(event, handleClickOutside));`
and
`["mousedown", "touchstart"].forEach(event => document. removeEventListener(event, handleClickOutside));
`

Reply:
Very good suggestion. I like it.